include/pmacc/particles/memory/buffers/ParticlesBuffer.hpp

```diff
@@ -196,12 +197,16 @@ namespace pmacc
             framesExchanges
                 ->addExchangeBuffer(receive, DataSpace<DIM1>(numFrameTypeBorders), communicationTag, true, false);
 
-            exchangeMemoryIndexer->addExchangeBuffer(
-                receive,
-                DataSpace<DIM1>(numFrameTypeBorders),
-                communicationTag | (1u << (20 - 5)),
-                true,
-                false);
+            /* Generate a new tag from this type
+             *
+             * The tag is the same each time this method is called (per instantiation of this template).
+             * Here it is fine, as there is the only instance object, and
+             * exchangeMemoryIndexer->addExchangeBuffer() requires the same tag on each call
+             */
+            using This = ParticlesBuffer<T_ParticleDescription, SuperCellSize_, T_DeviceHeap, DIM>;
+            auto const newTag = traits::GetUniqueTypeId<This, uint32_t>::uid();
```

Comment:
It should be possible to write `auto const newTag = traits::GetUniqueTypeId<ParticlesBuffer, uint32_t>::uid();`

Reply:
Ah, yes, I am always forgetting it